x-pack/plugins/security_solution/public/management/pages/trusted_apps/store/reducer.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// eslint-disable-next-line import/no-nodejs-modules
+import { parse } from 'querystring';
+import { matchPath } from 'react-router-dom';
+import { ImmutableReducer } from '../../../../common/store';
+import { AppLocation, Immutable } from '../../../../../common/endpoint/types';
+import { UserChangedUrl } from '../../../../common/store/routing/action';
+import { MANAGEMENT_ROUTING_TRUSTED_APPS_PATH } from '../../../common/constants';
+import { ListDataBindingChanged, TrustedAppsPageAction } from './action';
+import { TrustedAppsPageState } from '../state/trusted_apps_page_state';
+
+type StateReducer = ImmutableReducer<TrustedAppsPageState, TrustedAppsPageAction>;
+type CaseReducer<T extends TrustedAppsPageAction> = (
+  state: Immutable<TrustedAppsPageState>,
+  action: Immutable<T>
+) => Immutable<TrustedAppsPageState>;
+
+const initialPageState: TrustedAppsPageState = {
+  list: {
+    currentPage: { type: 'UninitialisedAsyncBinding' },
+    currentPageInfo: {
+      index: 1,
+      size: 10,
+    },
+  },
+  active: false,
+};
+
+const isTrustedAppsPageLocation = (location: Immutable<AppLocation>) => {
+  return (
+    matchPath(location.pathname ?? '', {
+      path: MANAGEMENT_ROUTING_TRUSTED_APPS_PATH,
+      exact: true,
+    }) !== null
+  );
+};
+
+const listDataBindingChanged: CaseReducer<ListDataBindingChanged> = (state, action) => {
+  return {
+    ...state,
+    list: {
+      ...state.list,
+      currentPage: action.payload.newBinding,
+    },
+  };
+};
+
+const userChangedUrl: CaseReducer<UserChangedUrl> = (state, action) => {
+  if (isTrustedAppsPageLocation(action.payload)) {
+    const query = parse(action.payload.search.slice(1));
+
+    return {
+      ...state,
+      list: {
+        ...state.list,
+        currentPageInfo: {
+          index: Number(query.page_index) || 1,
```

Comment:
This code here to get data out of the URL needs to be a little more robust. See this for an example:

https://github.com/elastic/kibana/blob/f286f8f6e1031e6ab1c0069c33ce7a8da69ac0ab/x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/selectors.ts#L127-L176

Because these value are coming from a user controlled location, some checks that should be done:

- Account for params appearing multiple times
- account for `NaN`
- account for invalid values (e.x. -1, 0, page_size=1000)

Reply:
Shared the implementation of endpoints list across this two tabs. I had to change the trusted apps implementation to a 0 based page indexes though. This should be fixed as part of the issue we have reported about 0 based page indexes in administration app.